### PR TITLE
fix(Filename): various fixes for Filename (used in FileUploader)

### PR DESCRIPTION
**Pull request by hellobrian**
resolves https://github.com/carbon-design-system/carbon-components-react/issues/122

See exact changes in the comments below, all other changes are a result of Prettier.

### CODE REVIEW

File: components/FileUploader.js [hunk 1]
```diff
@@ -83,46 +88,52 @@ class Filename extends Component {
...
-              name="checkmark--glyph"
-              description={`The file named, ${name}, has been uploaded.`}
-              style={style}
-              {...other}
-            />
-          : null}
-      </span>
-    );
+
+    if (status === 'uploading') {
```
Comment (hellobrian) [inline components/FileUploader.js]: __Changes:__ 
- Removed `<span>` which isn't used in the vanilla component
- Changed the overriding style for Loading component to make it `1rem` size

File: components/FileUploader.js [hunk 2]
```diff
@@ -83,46 +88,52 @@ class Filename extends Component {
...
+      return (
+        <div
+          className="bx--loading"
+          style={Object.assign(style, { width: '1rem', height: '1rem' })}
```
Comment (hellobrian) [inline components/FileUploader.js]: __Changes:__ Use new overriding style for Loading component to make it `1rem` size

File: components/FileUploader.js [hunk 3]
```diff
@@ -210,11 +221,15 @@ class FileUploader extends Component {
                   <span className="bx--file__state-container">
                     <Filename
                       status={filenameStatus}
-                      onClick={evt => this.handleClick(evt, index)}
+                      onClick={evt => {
+                        if (filenameStatus === 'edit') {
```
Comment (hellobrian) [inline components/FileUploader.js]: __Changes:__ Add conditional so that `onClick` is only fired when `filenameStatus === 'edit'`.

**Comment by marijohannessen** [comment]
👍 ✅

File: components/FileUploader.js [hunk 4]
```diff
@@ -114,39 +114,45 @@ class Filename extends Component {
...
+        >
+          <svg className="bx--loading__svg" viewBox="-42 -42 84 84">
+            <circle cx="0" cy="0" r="37.5" />
+          </svg>
+        </div>
+      );
+    } else if (status === 'edit') {
+      return (
+        <Icon
+          iconDescription={iconDescription}
```
Comment (asudoh) [inline components/FileUploader.js]: Getting the following trying out the change:

```
Warning: Unknown prop `iconDescription` on <svg> tag. Remove this prop from the element. For details, see https://fb.me/react-unknown-prop
    in svg (created by Icon)
    in Icon (created by Filename)
    in Filename (created by FileUploader)
    in span (created by FileUploader)
    in span (created by FileUploader)
    in div (created by FileUploader)
    in div (created by FileUploader)
    in FileUploader (created by Container)
    in div (created by Story)
    in div (created by Story)
    in Story (created by Container)
    in div (created by Container)
    in Container
```

Is there a upcoming corresponding change in `<Icon>`…? (Quickly checked `master` branch as well as PRs but couldn’t find one)

File: components/FileUploader.js [hunk 5]
```diff
@@ -114,39 +114,45 @@ class Filename extends Component {
...
+          className="bx--file-close"
+          name="close--glyph"
+          iconDescription={iconDescription}
```
Comment (asudoh) [inline components/FileUploader.js]: Seemingly a duplicate line of: https://github.com/carbon-design-system/carbon-components-react/pull/121/files#diff-e3c13a9b3a7ec4e1d7c64b976a5e419aR134

Comment (hellobrian) [inline components/FileUploader.js]: yeah that's right, I'll fix this. Thank you

File: components/FileUploader.js (hunk 4 above)
Comment (hellobrian) [inline components/FileUploader.js]: Hey, @asudoh, checking this now and I'm not seeing this error. It may have been there before but it's resolved now.

Comment (asudoh) [reply components/FileUploader.js]: OK thanks @hellobrian for checking!

**Comment by chrisdhanaraj** [comment]
Bluemix [toolchain](https://console.bluemix.net/devops/toolchains/6b8b7163-1ec8-4616-82fa-aa467a17af03 "storybook-toolchain-v2"): [Delivery Pipeline](https://console.bluemix.net/devops/pipelines/b84aaac8-8484-4a1e-902a-ebb526eb7998 "storybook-toolchain-v2") deployed [carbon-components-react](https://github.com/carbon-design-system/carbon-components-react/releases/tag/deploy-ibm-yp-us-south-chrisdhanaraj_org-Bluemix-Design-System-20171011-005532 "deploy-ibm-yp-us-south-chrisdhanaraj_org-Bluemix-Design-System-20171011-005532") to [Bluemix Design System](https://console.bluemix.net/apps/bc75a101-6472-481c-b48d-3eed1e9703f3?env_id=ibm:yp:us-south "ibm:yp:us-south:chrisdhanaraj_org:Bluemix Design System"), including commit 624a0cd3c2c95cbe215dd3ae4603fcfbc478fb0b

**Pull request merged**